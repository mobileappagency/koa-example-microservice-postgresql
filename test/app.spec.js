process.env.NODE_ENV = 'test';

const request = require('supertest');
const server = require('../app');
const knex = require('../lib/db/connection');

describe('routes : events', () => {

  beforeEach(() => {
    return knex.migrate.rollback()
      .then(() => { return knex.migrate.latest(); })
      .then(() => { return knex.seed.run(); });
  });

  afterEach(() => {
    return knex.migrate.rollback();
  });

  describe('POST /api/event', () => {
    it('should perform upsert on transaction duplicate', async () => {
      const response = await request(server.callback())
        .post('/api/event')
        .send({
          metadata: {
            transaction_id: 'qwertyuiop1234567890',
            created_at: new Date(),
            completed: true
          }
        });

      expect(response.status).toEqual(201);
      expect(response.body).toMatchSnapshot({
       data: [{ created_at: expect.any(String) }],
      });
    });

    it('should successfully insert a new event', async () => {
      const response = await request(server.callback())
        .post('/api/event')
        .send({
          metadata: {
            transaction_id: '1234567890qwertyuiop',
            created_at: new Date(),
            completed: false
          }
        });

      expect(response.status).toEqual(201);
      expect(response.body).toMatchSnapshot({
       data: [{ created_at: expect.any(String) }],
      });
    });
  });

  describe('GET /api/event/:tid', () => {
    it('should successfully retrieve an event by tid', async () => {
      const response = await request(server.callback())
        .get('/api/event/qwertyuiop1234567890');

      expect(response.status).toEqual(200);
      expect(response.body).toMatchSnapshot({
       data: [{ created_at: expect.any(String) }],
      });
    });
  });
});