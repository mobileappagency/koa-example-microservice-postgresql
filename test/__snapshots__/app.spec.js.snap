// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes : events GET /api/event/:tid should successfully retrieve an event by tid 1`] = `
Object {
  "data": Array [
    Object {
      "created_at": Any<String>,
    },
  ],
  "status": "Success.",
}
`;

exports[`routes : events POST /api/event should perform upsert on transaction duplicate 1`] = `
Object {
  "data": Array [
    Object {
      "created_at": Any<String>,
    },
  ],
  "status": "Success.",
}
`;

exports[`routes : events POST /api/event should successfully insert a new event 1`] = `
Object {
  "data": Array [
    Object {
      "created_at": Any<String>,
    },
  ],
  "status": "Success.",
}
`;
